### DISCUSSION TITLE
[TUTORIAL] patcher/loader guide for linux + Steam Play

### DISCUSSION THREAD

**Discussion by buck2202** [piotrulos/MSCModLoader#135] [General]
> [!IMPORTANT] 
> **There is way simplified proton tutorial if you click here -> #225** 
> Below one still probably works but may be overcomplicated

<details><summary>Click Here to show old tutorial</summary>
<p>

~~***Note: environment variable setup _possibly_ broken for Proton 6+. Needs investigation. Confirmed working on 5.x versions***~~
***Note: edited to add a new environment variable `STEAM_COMPAT_CLIENT_INSTALL_PATH` apparently needed for Proton 6/experimental. Comment if there are still issues***

Since MSC works fairly well through wine/proton on linux, I've been trying to get your loader functional as well. I found that versions <=1.1.7 worked without issue, but after you switched to doorstop, there was no indication of the mod loader being present. The patcher worked fine, but no console/settings/etc.

@piotrulos, I've read through past issues and some comments on steam/nexusmods/etc and know that you're (understandably) not interested in making code changes, but I did manage to get v.1.11 working successfully with only some minor configuration changes. It took a while to figure out so I just wanted to post here for others to find, or to offer it for your readme/wiki.

The bottom line is that there seems to be NOTHING inherently incompatible about the patcher/loader with wine/proton, i.e. "Steam Play". To successfully use it, we just have to make sure that the patcher operates in the same wine environment as the game, and set a dll override so that UnityDoorstop functions (again, in the proper wine environment).

I haven't tested many specific mods, but this is what worked for me to get the loader working

### Summary
- Identify your steam install path, proton version/install path, My Summer Car install path, and My Summer Car "compatdata" path
- Download and extract MSCModLoader, navigate a terminal to the extracted directory
- Set terminal variables
```
STEAMINSTALLDIR="YOUR STEAM CLIENT INSTALL DIRECTORY"
PROTONDIR="YOUR PROTON DIRECTORY"
MSCCOMPATDIR="YOUR MSC COMPATDATA DIRECTORY"
```
- Run patcher and install: `STEAM_COMPAT_CLIENT_INSTALL_PATH="${STEAMINSTALLDIR}" STEAM_COMPAT_DATA_PATH="${MSCCOMPATDIR}" WINEPREFIX="${MSCCOMPATDIR}/pfx" "${PROTONDIR}/proton" run ./MSCPatcher.exe`
- Open wine configuration for the MSC prefix: `STEAM_COMPAT_CLIENT_INSTALL_PATH="${STEAMINSTALLDIR}" STEAM_COMPAT_DATA_PATH="${MSCCOMPATDIR}" WINEPREFIX="${MSCCOMPATDIR}/pfx" "${PROTONDIR}/proton" run winecfg`
- Set library override for `winhttp` to `(builtin,override)`
- ...
- Profit

### Step by step instructions, with screenshots:
Some versions of Proton require an environment variable specifying the steam installation directory. When proton initializes a new game prefix, certain dlls are copied from this location (and they're confirmed and/or updated at every launch within the proton prefix). For most people, the directory we want is either `${HOME}/.steam/steam` or `${HOME}/.local/share/Steam`. My nearly fresh-install directory looks like this:
```
$ ls "${HOME}/.steam/steam"
appcache          depotcache                     linux32   servers                  steam_msg.sh                    ThirdPartyLegalNotices.html
bin               fossilize_engine_filters.json  linux64   ssfn___________________  steam.sh                        ubuntu12_32
bin_steamdeps.py  friends                        logs      steam                    steam_subscriber_agreement.txt  ubuntu12_64
bootstrap.tar.xz  GameOverlayRenderer64.dll      music     steamapps                steamui                         update_hosts_cached.vdf
clientui          graphics                       package   steamclient64.dll        tenfoot                         userdata
config            installscriptevalutor_log.txt  public    steamclient.dll          ThirdPartyLegalNotices.css
controller_base   legacycompat                   resource  steamdeps.txt            ThirdPartyLegalNotices.doc
```
and the `legacycompat` subdirectory should contain some steam-related dlls
```
$ ls "${HOME}/.steam/steam/legacycompat"
GameOverlayRenderer64.dll  iscriptevaluator.exe  steamclient64.dll  steamclient.dll  Steam.dll  SteamService.exe
```
If you get a directory not found error or empty listing, you probably have a non-standard install location and will have to track it down yourself. Something like `locate steamclient.dll` might help.

Next, identify the version of proton that you're using to launch the game. If you've set a specific override for MSC, it will be in the game's properties:
![Screenshot from 2021-02-24 06-52-45](https://user-images.githubusercontent.com/2069200/108996863-ee361780-766c-11eb-9cda-596483a49de2.png)
Otherwise, look to Steam>Settings>Steam Play
![Screenshot from 2021-02-24 06-47-46](https://user-images.githubusercontent.com/2069200/108996303-37399c00-766c-11eb-801f-5474a6302eac.png)
We need to find the installed directory for the correct version of proton. For `5.0-10` shown in my screenshots, you'd look in your steam library for "Proton 5.0". Open its properties, go to the local files tab, click browse, and a file browser should open to the correct path.
![Screenshot from 2021-02-24 06-55-45](https://user-images.githubusercontent.com/2069200/108997250-5c7ada00-766d-11eb-8a5a-c3101b3b40ee.png)
This is your "Proton directory"

Next, find your MSC installation directory. Go to MSC in your library, properties>local files>browse, and again, a file browser opens to the correct path.
![Screenshot from 2021-02-24 06-56-33](https://user-images.githubusercontent.com/2069200/108997339-79afa880-766d-11eb-844c-1d82faa81f4f.png)
This is your "MSC Install directory"

Finally, we need to find the compatdata directory for MSC. This is where the wine registry/libraries/etc live. If your MSC install directory was `/foo/bar/steamapps/common/My Summer Car`, the compatdata directory should be `/foo/bar/steamapps/compatdata/516750` (516750 is the game's App ID in Steam).
![Screenshot from 2021-02-24 07-00-06](https://user-images.githubusercontent.com/2069200/108997725-f3e02d00-766d-11eb-8633-7be0e63da746.png)
This is your "MSC compatdata directory"

Let's set the paths that we've identified into variables for easier use. Open a terminal, and enter
```
STEAMINSTALLDIR="YOUR STEAM CLIENT INSTALL DIRECTORY"
PROTONDIR="YOUR PROTON DIRECTORY"
MSCCOMPATDIR="YOUR MSC COMPATDATA DIRECTORY"
```
so, for me, it's
```
STEAMINSTALLDIR="${HOME}/.steam/steam"
PROTONDIR="/media/evodata/SteamLibrary.linux/steamapps/common/Proton 5.0"
MSCCOMPATDIR="/media/evodata/SteamLibrary.linux/steamapps/compatdata/516750"
```
Put all paths in quotes. These variables are not directly accessed by proton and don't need to be exported; they're only for our convenience to avoid retyping long paths (and allow you to copy/paste the commands below). They ONLY APPLY TO THE TERMINAL WINDOW YOU TYPE THEM IN, so we're going to use this window for everything. Download and extract the MSCLoader zip wherever you want, and navigate to it in the terminal you have open.
![1--paths](https://user-images.githubusercontent.com/2069200/134576853-5e546233-062d-497b-bfd4-fa6662fe7ef9.png)

Run the patcher with this command:
```
STEAM_COMPAT_CLIENT_INSTALL_PATH="${STEAMINSTALLDIR}" STEAM_COMPAT_DATA_PATH="${MSCCOMPATDIR}" WINEPREFIX="${MSCCOMPATDIR}/pfx" "${PROTONDIR}/proton" run ./MSCPatcher.exe
```
![Screenshot from 2021-02-24 16-11-49](https://user-images.githubusercontent.com/2069200/109066550-35e39000-76bb-11eb-9a99-b5244df20aa0.png)
A good hint that you're on the right track here is that the patcher detects your "My Documents" folder as belonging to "steamuser". Click the "MSC Folder" button and navigate to your My Summer Car install folder. In the proton environment that we're using, your "real" filesystem is under drive letter `Z` (So, `Z:\media\evodata\SteamLibrary.linux\steamapps\common\My Summer Car` for me). Click "Install MSCLoader".

Choosing "My Documents" for your mod folder corresponds to `<MSC compatdata>/pfx/drive_c/users/steamuser/My Documents/MySummerCar/Mods` in your "real" filesystem

Good so far?
![Screenshot from 2021-02-24 16-23-26](https://user-images.githubusercontent.com/2069200/109067536-a3dc8700-76bc-11eb-84f6-aebe5028b3fa.png)

Close the patcher window (it cannot launch the game, and we're not done yet), go back to your terminal, and enter the following command
```
STEAM_COMPAT_CLIENT_INSTALL_PATH="${STEAMINSTALLDIR}" STEAM_COMPAT_DATA_PATH="${MSCCOMPATDIR}" WINEPREFIX="${MSCCOMPATDIR}/pfx" "${PROTONDIR}/proton" run winecfg
```
In the "Libraries" tab, check your "Existing overrides" for `winhttp`. If it isn't present, type `winhttp` it into the "New override for library" box and click "Add". In the end, you should have a `winhttp (native,builtin)` line in your overrides
![Screenshot from 2021-02-24 16-30-04](https://user-images.githubusercontent.com/2069200/109068304-ad1a2380-76bd-11eb-9b83-e6f0e7c7fd55.png)
If it doesn't say `(native,builtin)`, just highlight `winhttp`, click "Edit", and select "Native then builtin"
![Screenshot from 2021-02-24 16-34-12](https://user-images.githubusercontent.com/2069200/109068709-3a5d7800-76be-11eb-844d-dd1fa1ff5386.png)
Click apply, and OK, and the "Wine Configuration" window should go away.

What did we just do? MSC Mod Loader uses Unity Doorstop to inject itself; Doorstop overrides a system dll with a placeholder that links to the "real" library _plus_ a handle to additional code. When a Unity game looks for a library on Windows, it first looks in the game's root directory. If it's found there, it uses that one; if not, it falls back to the system version. On wine/proton, this _does not happen_ by default. Generally this is good and required (wine provides many "system" dlls, which is what makes it work at all), but in this case, it prevents the Doorstop hook from being seen when the game launches. By overriding `winhttp`, we've just made the proton behavior match what happens on Windows.

We're done with the terminal, so you can close it. Launch the game through steam as you normally would, and you should be all set
![proof](https://user-images.githubusercontent.com/2069200/134578389-d0cd2c68-e3fd-4918-b21f-138ba8ad4287.png)

</p>
</details>

**Comment by piotrulos** [comment]
Good tutorial, I leave it pinned if someone wants it

**Comment by buck2202** [comment]
![Screenshot from 2021-02-24 20-32-01](https://user-images.githubusercontent.com/2069200/109089706-abf9ee00-76df-11eb-9ac8-919de6dd2d31.png)

Just a quick followup to say I've tried a handful of mods and so far everything has worked as it should. Hope this is helpful to someone. Thanks for your work on this project!

**Comment by slowpoke9292** [comment]
how do i fix crashing? 
[output_log.txt](https://github.com/piotrulos/MSCModLoader/files/6178371/output_log.txt)

**Comment by buck2202** [comment]
I don't see anything that looks like a crash in your output. Can you confirm that you applied the patch using the game's proton prefix, and that you're launching the game through steam? Which version of proton?

**Comment by slowpoke9292** [comment]
> I don't see anything that looks like a crash in your output. Can you confirm that you applied the patch using the game's proton prefix, and that you're launching the game through steam? Which version of proton?
I got it fixed using windows steam, plus I don't have Proton. Thank you though :)
![Screenshot 2021-03-22 2 01 48 PM](https://user-images.githubusercontent.com/80487507/112036624-3daa1f00-8b17-11eb-89c8-776de0d0fd4f.png)

**Comment by slowpoke9292** [comment]
(I can also post an ingame screenshot if needed.)

**Comment by sharkjung** [comment]
Can't get anything upwards of 1.1.7 to work. As you stated above, the patcher works fine but the mod loader simply doesn't even seem to be installed. I was using Proton 5.13 so far, but tried version 5.0-10 in my last attempt and that gave me the same results. Could you help me out?

**Comment by buck2202** [comment]
> Can't get anything upwards of 1.1.7 to work. As you stated above, the patcher works fine but the mod loader simply doesn't even seem to be installed. I was using Proton 5.13 so far, but tried version 5.0-10 in my last attempt and that gave me the same results. Could you help me out?

@petralka that sounds like the necessary `winhttp` override isn't set up. did you go through the whole process using the game's proton prefix as described above? you can always verify local files in steam to get a fresh start if you think there's a chance you were working in your pc's default wine for any of it

edit: even though the patcher might seem like it's working if you just use the system's default wine installation, I'd recommend doing everything under the correct proton prefix so that the patcher sees the same paths as when you launch the game through steam

**Comment by slowpoke9292** [comment]
> > Can't get anything upwards of 1.1.7 to work. As you stated above, the patcher works fine but the mod loader simply doesn't even seem to be installed. I was using Proton 5.13 so far, but tried version 5.0-10 in my last attempt and that gave me the same results. Could you help me out?
> 
> @petralka that sounds like the necessary `winhttp` override isn't set up. did you go through the whole process using the game's proton prefix as described above? you can always verify local files in steam to get a fresh start if you think there's a chance you were working in your pc's default wine for any of it
> 
> edit: even though the patcher might seem like it's working if you just use the system's default wine installation, I'd recommend doing everything under the correct proton prefix so that the patcher sees the same paths as when you launch the game through steam


An update to my last comment: I got it working with the full package.

**Comment by sharkjung** [comment]
I tried the full package this time, one thing I noticed was that I wrote `MSCOMPATDIR` instead of `MSCCOMPATDIR`, going through my command history I discovered that I was using both names.  Anyways, it worked! Thanks for the guide, @buck2202!

**Comment by Kreyren** [comment]
Setting the `winhttp` override fixed the deployment on my system, thanks

**Comment by thebirdwashere** [comment]
I am getting this error when running the `STEAM_COMPAT_DATA_PATH="${MSCCOMPATDIR}" WINEPREFIX="${MSCCOMPATDIR}/pfx" "${PROTONDIR}/proton" run ./MSCPatcher.exe` command.

`Traceback (most recent call last):
  File "/home/jacob/.local/share/Steam/steamapps/common/Proton 6.3//proton", line 1264, in <module>
    g_session.init_session(sys.argv[1] != "runinprefix")
  File "/home/jacob/.local/share/Steam/steamapps/common/Proton 6.3//proton", line 1078, in init_session
    g_compatdata.setup_prefix()
  File "/home/jacob/.local/share/Steam/steamapps/common/Proton 6.3//proton", line 652, in setup_prefix
    steamdir = os.environ["STEAM_COMPAT_CLIENT_INSTALL_PATH"]
  File "/usr/lib/python3.9/os.py", line 679, in __getitem__
    raise KeyError(key) from None
KeyError: 'STEAM_COMPAT_CLIENT_INSTALL_PATH'
`
Does anyone know what is wrong? I tried re setting the variables, and that didnt help, i also started the game for the first time to see if that would help.

**Reply by Kreyren** [reply]
That is an error from Proton not WineHQ which seems to be looking for `STEAM_COMPAT_CLIENT_INSTALL_PATH` ?

**Reply by thebirdwashere** [reply]
Looks like it. Not sure why it would be looking for it since it was specified in the command.

**Reply by Kreyren** [reply]
i don't see it defined in the command you sent, but i assume that you forgot to export

```console
$ PROTONDIR="YOUR PROTON DIRECTORY"
$ MSCCOMPATDIR="YOUR MSC COMPATDATA DIRECTORY"
```

?

I've noticed that the original post is forgetting about `export` which is important for this runtime so consider doing:

```console
$ export PROTONDIR="YOUR PROTON DIRECTORY"
$ export MSCCOMPATDIR="YOUR MSC COMPATDATA DIRECTORY"
```

To make those variables available to the command.

**Reply by thebirdwashere** [reply]
Ill try adding export, before i was just doing PROTONDIR="YOUR PROTON DIRECTORY" and the same for MSCCOMPATDIR

**Reply by Kreyren** [reply]
Just for completeness you replaced the `YOUR PROTON DIRECTORY` with the actual path right? e.g. `/home/someone/.../`

**Reply by thebirdwashere** [reply]
Seems like adding to those commands that still didnt fix it, do i need to add it to the command to launch the patcher?

**Reply by Kreyren** [reply]
i would try a full path to the patcher e.g. `/home/someone/dir/MSCPatcher.exe` instead of using `./MSCPatcher.exe`

**Reply by Kreyren** [reply]
or make sure to `cd` in the MySummerCar directory and have it the same as where the patcher is located.

**Reply by thebirdwashere** [reply]
Still getting the error about install path

**Reply by thebirdwashere** [reply]
> or make sure to `cd` in the MySummerCar directory and have it the same as where the patcher is located.

So move the patcher to the MySummerCar dir?

**Reply by Kreyren** [reply]
Is the `STEAM_COMPAT_DATA_PATH="${MSCCOMPATDIR}" WINEPREFIX="${MSCCOMPATDIR}/pfx" "${PROTONDIR}/proton" run winecfg` working? Should open a window alike

![image](https://user-images.githubusercontent.com/11302521/130710721-e2d46284-690e-4abe-907e-6c248baaf0f8.png)

**Reply by thebirdwashere** [reply]
> Is the `STEAM_COMPAT_DATA_PATH="${MSCCOMPATDIR}" WINEPREFIX="${MSCCOMPATDIR}/pfx" "${PROTONDIR}/proton" run winecfg` working? Should open a window alike
> 
> ![image](https://user-images.githubusercontent.com/11302521/130710721-e2d46284-690e-4abe-907e-6c248baaf0f8.png)

No, that wont open either.

**Reply by Kreyren** [reply]
Try just `"${PROTONDIR}/proton" run winecfg` if that doesn't work then your proton is broken

**Reply by thebirdwashere** [reply]
> Try just `"${PROTONDIR}/proton" run winecfg` if that doesn't work then your proton is broken

Seems it's broken, Should I try an older version such as the one in the screenshots?

**Reply by Kreyren** [reply]
i would start by checking integrity assuming that you are using steam then different version

**Reply by Kreyren** [reply]
Optionally consider using just winehq with disabled steam dependency which on my system gets more efficient runtime.

**Reply by thebirdwashere** [reply]
Seems that switching to Proton 5.13 allowed me to launch winecfg and the patcher, now to try and see if it actually patched.
Yup, it patched, thank you for the help! I did check the integrity first and nothing showed up.

**Reply by buck2202** [reply]
`PROTONDIR` and `MSCCOMPATDIR` are only used within the scope of a single terminal session to avoid retyping long paths, so it shouldn't be necessary to export them; they're just variable names that I chose, not things that I know are used internal to proton. They'll just get expanded by the shell in the patcher and winecfg commands.

I don't have access to a proton 6.3 install to check at the moment, but it's possible the environment variables that it needs are different than the version I tested with. Can you check a 5.x version and I'll try to circle back to check against 6.x?
edit: seems you did that already! I've had lots of games just outright fail to launch with proton 6, so I've been avoiding it

**Reply by Kreyren** [reply]
Just to clarify:

```console
[kreyren@leonid:~]$ somevar=whee

[kreyren@leonid:~]$ echo $somevar
whee

[kreyren@leonid:~]$ echo 'echo "$somevar"' > test.sh

[kreyren@leonid:~]$ cat test.sh 
echo "$somevar"

[kreyren@leonid:~]$ chmod +x test.sh

[kreyren@leonid:~]$ ./test.sh  

[kreyren@leonid:~]$ somevar=whooo ./test.sh 
whooo
```

If you need `PROTONDIR` and `MSCCOMPATDIR` accesible to the command during runtime then you have to `export` else passing them as a variable to specify proton-related environment vars works without the command having access to them.

**Reply by Kreyren** [reply]
@buck2202 i recommend mentioning inconsistency with proton and referring to this issue since it seems to be a common problem with proton on steam.

**Reply by buck2202** [reply]
@Kreyren yep, what I meant was that `MSCCOMPATDIR` and `PROTONDIR` are just for convenience; they don't need to be accessible inside the runtime. Proton internally needs `STEAM_COMPAT_DATA_PATH` and `WINEPREFIX`, the shell expands `MSCCOMPATDIR` into its constant value when we use it to define the others, and `${PROTONDIR}/proton` is just the complete path to the executable.

```
$ echo 'echo "$inside"' >test.sh
$ chmod +x ./test.sh 
$ cat ./test.sh
echo "$inside"
$ echo "$inside"

$ ./test.sh 

$ outside="asdf"
$ inside="${outside}" ./test.sh 
asdf
```
this is all we're really doing. 

I'll add a note about possible incompatibility with proton 6, and try to come up with a solution.

**Comment by Kreyren** [comment]
I don't think that this is an incompatibility with proton 6 rather steam is unreliable in providing a sane proton sometimes.


Sent from ProtonMail mobile



\-------- Original Message --------
On Aug 25, 2021, 20:03, David B < ***@***.***> wrote:

>
>
>
> ***@***.***[Kreyren] yep, what I meant was that `MSCCOMPATDIR` and `PROTONDIR` are just for convenience; they don't need to be accessible inside the runtime. Proton internally needs `STEAM_COMPAT_DATA_PATH` and `WINEPREFIX`, the shell expands `MSCCOMPATDIR` into its constant value when we use it to define the others, and `${PROTONDIR}/proton` is just the complete path to the executable.
>
> ```
> $ echo 'echo "$inside"' >test.sh
> $ chmod +x ./test.sh
> $ cat ./test.sh
> echo "$inside"
> $ echo "$inside"
>
> $ ./test.sh
>
> $ outside="asdf"
> $ inside="${outside}" ./test.sh
> asdf
> ```
>
> this is all we're really doing.
>
> I'll add a note about possible incompatibility with proton 6, and try to come up with a solution.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, [view it on GitHub][], or [unsubscribe][].![ACWHM6KORYAK4XYF5YTXV2LT6UWA5A5CNFSM5CXZU6VKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAJNOBQ.gif][]


[Kreyren]: https://github.com/Kreyren
[view it on GitHub]: https://github.com/piotrulos/MSCModLoader/discussions/135#discussioncomment-1234694
[unsubscribe]: https://github.com/notifications/unsubscribe-auth/ACWHM6OVQ6C2SIAFVZBOMDDT6UWA5ANCNFSM5CXZU6VA
[ACWHM6KORYAK4XYF5YTXV2LT6UWA5A5CNFSM5CXZU6VKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAJNOBQ.gif]: https://github.com/notifications/beacon/ACWHM6KORYAK4XYF5YTXV2LT6UWA5A5CNFSM5CXZU6VKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAJNOBQ.gif

**Comment by piotrulos** [comment]
Just a thing that I didn't see anyone metion, Doorstop also supports `version.dll` proxy, that should be more compatible with wine stuff.
If you install latest modloader (1.1.16) just rename `winhttp.dll` file to `version.dll` and see if that fixes issues.

**Reply by Kreyren** [reply]
I don't think it's incompatibility issue rather wine configuration since we have FLOSS rewrites of all standard libraries and wine by default uses those instead of the one provided by programs and alike. So for wine to use `winhttp.dll` or `version.dll` it has to be set as `native`

**Comment by Greasy2071** [comment]
Hi. 
I've done everything on the guide to a T, and redid it too, but for some reason the mod loader doesn't show up? There's a text file about the loader crashing which i'll attach 
[MSCLOADER_CRASH.txt](https://github.com/piotrulos/MSCModLoader/files/7112194/MSCLOADER_CRASH.txt)
Thanks in advance for any help or advice.

**Reply by buck2202** [reply]
hm, can you paste the contents of `steamapps/common/My Summer Car/doorstop_config.ini`? I'm wondering if there's just something non-standard in there. with that said, I haven't updated to the latest mod loader in a while, so something may have changed

**Reply by buck2202** [reply]
without seeing it, I'd try deleting any comments from that file, as they seem to be causing a parse error

**Reply by piotrulos** [reply]
It doesn't seem to be a valid config, since official config file doesn't have comments.    
It looks more like config file from "proloader" which is totally incompatible with regular loader.

**Reply by Greasy2071** [reply]
```ini
# General Doorstop settings
[UnityDoorstop]
# Specifies whether assembly executing is enabled
enabled=true
# Specifies the path (absolute, or relative to the game's exe) to the DLL/EXE that should be executed by Doorstop
targetAssembly=mysummercar_Data\Managed\MSCLoader.dll
# Specifies whether Unity's output log should be redirected to <current folder>\output_log.txt
redirectOutputLog=true
# If enabled, DOORSTOP_DISABLE env var value is ignored
# USE THIS ONLY WHEN ASKED TO OR YOU KNOW WHAT THIS MEANS
ignoreDisableSwitch=false
```

alright i've took all the comments out (the bits with the hashtags i hope lol) havent saved changes yet in case i messed something up which is very likely considering all my brain cells went out of me when i was born save one
sorry for the long reply my emails don't come up unless i check them myself. 
i did have the mod loader pro but i haven't used it since i got the latest version of this one. 
thanks for replying 
edit: oh wow it makes the comments big 
edit 2: ok i deleted the comments and launched the game and MOP is throwing a bunch of errors. I'll try to fix that maybe reinstall it or smth idk but the loader itself is on the menu of the game 
//Fixed formatting for you

**Comment by Greasy2071** [comment]
here are the console errors
![516750_20210907121235_1](https://user-images.githubusercontent.com/90154379/132335516-816d35fd-092c-49f7-8589-b28edc66f002.png)
not sure if you'll be able to see them well

**Reply by piotrulos** [reply]
Download MOP 3.2.3 [https://www.nexusmods.com/mysummercar/mods/146?tab=files&file_id=1556](https://www.nexusmods.com/mysummercar/mods/146?tab=files&file_id=1556) if you want a working version.   
3.3 doesn't work.

**Comment by piotrulos** [comment]
I think I may have solution for linux in 1.2 modloader update, but no promises yet since it's untested. It's really low priority, and it will be released as completly "experimental" thing.

**Comment by buck2202** [comment]
I've updated the original post to hopefully resolve setup issues in Proton 6/experimental. I've tested the steps above for 1.1.16 loader with proton experimental as of today, and the errors @grumpzalt saw above should be covered (undefined `STEAM_COMPAT_CLIENT_INSTALL_PATH`). The changes seem harmless for Proton 5

**Comment by c4tgrl** [comment]
I get a KeyError 'STEAM_COMPAT_CLIENT_INSTALL_PATH

**Reply by buck2202** [reply]
can you either paste your terminal output in a code block, or add a (terminal) screenshot?

**Comment by AbsoluteAnton** [comment]
When i try to choose the msc folder in the installer i cant find the folder.
It is located in /home/Anton/.local/share/Steam/steamapps/common/My Summer Car/.
The problem is that the folder ".local" isnt listed in the wine file explorer. 
Is it the only way to move the whole steam directory? @buck2202

**Reply by buck2202** [reply]
oh, that's an annoying thing I didn't know about the wine file explorer

You should be able to navigate to `/home/Anton`, type `.local` in the file name box+hit enter, then continue browsing. Alternately, you could just symlink the directory to some other non-hidden location

You shouldn't have to move anything, though

**Reply by AbsoluteAnton** [reply]
Ok, thanks. I'll try

**Reply by AbsoluteAnton** [reply]
I got it to work.
Thank you!

**Comment by AbsoluteAnton** [comment]
If this is irrelevant you can delete it but could you smart guys post a tutorial on how to get msc editor up and working?

**Reply by buck2202** [reply]
been I while since I've played/tried to use it, but if I remember correctly, the editor mostly "just worked" with standard Wine. I think the interactive map was iffy, but editing fields worked ok. don't recall if there was a trick to it, though, sorry :/

if just plain-old wine doesn't work, might just wanna try its nexusmods discussion or github (linked from its nexus page). if I get back into it, though, I'll look for your post and follow-up

**Comment by BerkeSinanYetkin** [comment]
You know, adding the .exe as a non-steam game and starting it through steam with proton also works

**Comment by xbelanch** [comment]
At the moment, I followed all the steps but with Proton 8.0 but no luck at the end. Any hints to solve this?

Five minutes later: it works (typo error -> winehttp instead of winhttp)

**Comment by MillionsToOne** [comment]
I think there should be a disclaimer for those that have MSC installed in another path that they need to replace the path inside the msc.sh files to the path matching  the game directory's location, just to make it more noob friendly since they might not realize it.

**Comment by sofiedotcafe** [comment]
You could just use protontricks?

**Reply by MillionsToOne** [reply]
That isn't in the official guide, which a lot of people follow

**Comment by BerkeSinanYetkin** [comment]
Wait what there is native support for linux? Since when?

**Reply by MillionsToOne** [reply]
It's buggy. Don't bother with it.

**Comment by HEPOSHEIKKI** [comment]
Fix for those getting the `STEAM_COMPAT_CLIENT_INSTALL_PATH KeyError`:

Add `STEAM_COMPAT_CLIENT_INSTALL_PATH="~/.steam/steam"` in front of any command giving the error.

**Comment by realairacobra** [comment]
Game says checking for updates and completely freezes; before you say something about me needing to buy the game, i have it, it's just not detecting my steam account.
![image](https://github.com/piotrulos/MSCModLoader/assets/93622023/602aeddb-a131-4162-ac68-a6206b3fd0c7)
[output_log.txt](https://github.com/user-attachments/files/15981127/output_log.txt)

![image](https://github.com/piotrulos/MSCModLoader/assets/93622023/b43e2d4c-3caf-4fbd-9d21-566947fe74f8)


Fixed by completely reinstalling the game

**Reply by MillionsToOne** [reply]
You gave me the minimal info possible. Which proton version did you use? Did you follow all the steps in this tutorial correctly? Which distro do you use? Did you run Steam in flatpak?

**Reply by realairacobra** [reply]
did this kinda rushed, my bad, expected output_log to give out some of that info, im using steam from the AUR, i tried proton GE and a bunch of other versions but its all the same, im on arcolinux, i followed everything correctly

**Reply by MillionsToOne** [reply]
> did this kinda rushed, my bad, expected output_log to give out some of that info, im using steam from the AUR, i tried proton GE and a bunch of other versions but its all the same, im on arcolinux, i followed everything correctly

Steam isn't in AUR. Only in the extra repository. Regardless, you should try Proton 8.0-5. Try reinstalling the mod loader, since the log suggests you didn't install it correctly. Most likely you used the wrong STEAM_COMPAT_CLIENT_INSTALL_PATH value in it. The right path is /home/$USER/.steam/steam/

**Reply by realairacobra** [reply]
ah alright, im trying to get it running on 6.3-8 because thats where force feedback works for me on other games, STEAM_COMPAT_CLIENT_INSTALL_PATH is ${HOME}/.steam/steam which leads to the directory without any issues, i'll try the one you sent anyway

**Reply by realairacobra** [reply]
Compeletely deleted and reinstalled msc it works now

**Reply by MillionsToOne** [reply]
> ah alright, im trying to get it running on 6.3-8 because thats where force feedback works for me on other games, STEAM_COMPAT_CLIENT_INSTALL_PATH is ${HOME}/.steam/steam which leads to the directory without any issues, i'll try the one you sent anyway

Well yes, $HOME is short for /home/$USER

**Comment by Visne** [comment]
You can bypass all this by setting launch options in Steam to `WINEDLLOVERRIDES="winhttp=n,b" %command%`.

**Reply by AliceSchaffernoth** [reply]
Nothing was working until I did this. Thank you!

**Comment by Mircas001** [comment]


**Reply by Mircas001** [reply]
this might be related to the new 1.3 installer

**Reply by piotrulos** [reply]
There is new tutorial, way simpler: #225

**Reply by Mircas001** [reply]
OMG THANK YOU
